libdevcore/JsonUtils.h

```diff
@@ -24,4 +24,25 @@ namespace dev
 {
 // Throws UnknownField() if _obj contains field names not listed in _allowedFields.
 void validateFieldNames(json_spirit::mObject const& _obj, std::set<std::string> const& _allowedFields);
+
+// Converts json value type to string
+std::string jsonTypeAsString(json_spirit::Value_type _type);
+
+enum jsonField
+{
+    Required,
+    Optional
+};
+using jsonTypeSet = std::set<json_spirit::Value_type>;
+using jsonType = std::pair<jsonTypeSet, jsonField>;
```

Comment:
Start type name with capital letter, also this is something like `JsonTypesAndPresence` I'd say